wsproxy: port to _oasis and ppx

mseri opened the pull request:
Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

Commit:
wsproxy: port to _oasis and ppx

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

gaborigloi commented:
Would it be better to auto-generate all the OASIS build files?

lindig commented:
I like this. I also wonder, should we just add the `_oasis` file and rely on the user (and the RPM spec file) to call `oasis setup`?

Commit:
lwt_support: use bind after close and connect

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

Commit:
ocp-indent code-base

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

Commit:
reasonably re-indent docstrings

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

mseri commented:
I agree with both of you. I think we should rely on opam, rpmbuild or the user doing the right thing.

Commit:
remove oasis generated files

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

lindig commented on src/wsproxy.ml:
```diff
@@ -0,0 +1,108 @@
+  in
+  loop ()
+
+let ($) f x = f x
```
This is a Haskell-ism but in OCaml it associates to the left, which is wrong. That is why `@@` is used for this in OCaml, as it associates to the right. ~~I also don't see it being used~~

Commit:
remove unnecessary $ operator

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

Commit:
backport fix from v1.0.0

commits 59879b3d052 and deb02fdbfb1 titled 'remove base64encoding and fix leaking process'

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

Commit:
Remove unnecessary use of random from the backport

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

lindig commented:
This looks good but I would like to see a review in addition b someone who knows the history of this package.

Commit:
wsproxy: do not use Versioned Lwt_unix bind, it has appeard in a too recent version of lwt

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

Commit:
fix merge conflict

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

gaborigloi commented:
Apparently the `setup.ml` file is still in the repo, but this is an auto-generated file that can be created with `oasis setup` - was this file kept deliberately in the repo?

mseri commented:
No, it remained by mistake when I removed the oasis generated files. I am happy to make a pr to get rid of it or to wait for the time we remove the autogenerated files from the other repos

mseri commented:
Here we go https://github.com/xapi-project/wsproxy/pull/4
